ganga/GangaCore/GPIDev/Base/Proxy.py

```diff
@@ -1205,6 +1205,165 @@ def _getattribute(self, name):
     return type(name, (GPIProxyObject,), d)
 
 
+import copy
+
+primitive_types = (int, bool, float, long, complex)
+sequence_types = (str, unicode, list, tuple, bytearray, buffer, xrange)
+
+
+class COWProxy(object):
+    """
+    Copy On Write Proxy
+
+    The proxy is a transparent wrapper that intercepts all access to the
+    underlying object. When first created, it creates a shallow copy
+    of the underlying object and behaves like a shadow reference. When
+    any attribute of the underlying object is modified, the COWProxy either
+    creates a complete deep copy (it deep copies the entire object it was
+    tracking) or creates a partial deep copy (it deep copies only the
+    attribute that was touched). This behavior is triggered by an appropriate
+    flag.
+    """
+    __slots__ = ["_obj", "__weakref__", "__slots__", "_is_copied",
+                 "_enable_partial_copy", "_attr_map"]
+
+    _is_copied = False
+    _attr_map = {}
+
+    def __init__(self, obj, _partial_copy=True):
+        object.__setattr__(self, "_obj", obj)
+        object.__setattr__(self, "_enable_partial_copy", _partial_copy)
```

Comment:
as long as the attribute names `_obj` and `_enable_partial_copy` are defined within `__slots__` (as they are) you should be able to assign these using the standard:
```python
self._obj = obj
self._enable_partial_copy = _partial_copy
```
rather than this unbounded (static) call to object base class's `__setattr__`.

Reply:
I see it's because you have overridden the `object.__setattr__` and you don't want it triggered on these instance variables